### PR TITLE
Fixes #64 Multiline variables

### LINKED ISSUE #64: Doesn't support multiline vars
If I set something like this:

```env
KEY='this is
a multiline
variable'
```

`godotenv` returns:
```console
Can't separate key from value
```

Even though that's a valid variable assignment.

I'm currently at work; will attempt a fix later this week.

### PR DESCRIPTION
**Pull request by esttorhe**
# Fixes #64 

This `PR` handles multi-line values by single quote delimiter `'`.

The parsing of the whole `.env` file is pretty naive (as expressed in the `README`) but does the work. I tried to add as little code as possible and change as little as possible just to support this.

All tests are passing and 3 new ones were added (1 success and 2 different failing scenarios)

### CODE REVIEW

**Comment by esttorhe** [comment]
Hold on reviewing; I identified an edge case that needs to be addressed first

**Comment by kiambogo** [comment]
Any update on this? Also experiencing this issue. What was the edge case you found?

**Comment by esttorhe** [comment]
@kiambogo same use case as me; that's why I added the support and noticed the issue.

I think the naive implementation could still be used for a first version just requiring the use of either `"` or `'` to signal the beginning of a multiline; after that we can add some better support for this to remove the naive implementation for a more solid one.

**Comment by esttorhe** [comment]
@joho Any chance to get a review?

File: godotenv.go [hunk 1]
```diff
@@ -209,7 +246,7 @@ func readFile(filename string) (envMap map[string]string, err error) {
 	return Parse(file)
 }
 
-func parseLine(line string, envMap map[string]string) (key string, value string, err error) {
+func parseLine(line string, envMap map[string]string, isMultiline bool) (key string, value string, multilineValue bool, err error) {
```
Comment (joho) [inline godotenv.go]: I'd appreciate if you could find a way to restructure this to not rely on a boolean flag as I'm in Fowler's camp on the topic https://www.martinfowler.com/bliki/FlagArgument.html

File: godotenv.go [hunk 2]
```diff
@@ -109,17 +109,54 @@ func Parse(r io.Reader) (envMap map[string]string, err error) {
...
-			key, value, err = parseLine(fullLine, envMap)
+			var isMultilineValue bool
+			key, value, isMultilineValue, err = parseLine(fullLine, envMap, multilineValue)
 
 			if err != nil {
 				return
 			}
-			envMap[key] = value
+
+			if multilineValue == true && isMultilineValue == false && len(key) <= 0 {
```
Comment (joho) [inline godotenv.go]: I have to admit I'm really confused by the two multilineValue booleans, and I'm not sure if that's a structural problem or a naming problem of the variables.

Comment (esttorhe) [reply godotenv.go]: Yeah; this is a fair comment. I'm not particularly happy with the implementation and thus wanted to get some feedback from you since is your codebase.

I'll close this PR and will refactor in a cleaner way probably this weekend.

🙇
